vignettes/cli-config-user.Rmd

```diff
@@ -9,18 +9,83 @@ editor_options:
     wrap: sentence
 ---
 
-These are environment variables and options that uses may set, to modify
+These are environment variables and options that users may set, to modify
 the behavior of cli.
 
-## User facing environment variables
+## Hyperlinks
+
+cli uses ANSI escape sequences to create hyperlinks.
+Specifically, cli creates [OSC 8 hyperlinks](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda) that have this syntax:
+
+```
+OSC 8 ; {OPTIONAL PARAMS } ; {URI} ST {LINK TEXT} OSC 8 ; ; ST
+```
+
+Under the hood, [style_hyperlink()] is the helper that forms these links, but it is more common to request them indirectly via inline markup (documented in `help("links")`).
 
 ### `R_CLI_HYPERLINK_MODE`
 
 Set to `posix` to force generating POSIX compatible ANSI hyperlinks.
+This is about the specific operating system command (OSC) and string terminator (ST) used in hyperlinks.
+
 If not set, then RStudio compatible links are generated. This is a
 temporary crutch until RStudio handles POSIX hyperlinks correctly, and
 after that it will be removed.
 
+### `cli.hyperlink` option and `R_CLI_HYPERLINKS` env var
+
+SIDE NOTE: THE "S" AT THE END R_CLI_HYPERLINKS IS REGRETTABLE, IN TERMS OF CONSISTENCY WITH RELATED CONFIG.
```

Comment:
This has tripped me up multiple times in this little project 😬 

Any chance you want to create `R_CLI_HYPERLINK` (no S)?

Reply:
I don't really want to, but a PR is welcome. We would still need to fall back to `R_CLI_HYPERLINKS` of course, for compatibility.

Btw. I am not sure if these notes are useful for users, especially in a reference, where we would ideally document what happens, and not why.